python/cudf/cudf/options.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+from collections.abc import Container
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, Optional
+
+
+@dataclass
+class Option:
+    default: Any
+    value: Any
+    description: str
+    validator: Callable
+
+
+_OPTIONS: Dict[str, Option] = {}
+
+
+def _register_option(
+    name: str, default_value: Any, description: str, validator: Callable
+):
+    """Register an option.
+
+    Parameters
+    ----------
+    name : str
+        The name of the option.
+    default_value : Any
+        The default value of the option.
+    description : str
+        A text description of the option.
+    validator : Callable
+        Called on the option value to check its validity. Should raise an
+        error if the value is invalid.
+    """
+    validator(default_value)
+    _OPTIONS[name] = Option(
+        default_value, default_value, description, validator
+    )
+
+
+def get_option(name: str) -> Any:
+    """Get the value of option.
+
+    Parameters
+    ----------
+    key : str
+        The name of the option.
+
+    Returns
+    -------
+    The value of the option.
+    """
+    return _OPTIONS[name].value
```

Comment:
Need to document in the docstring that this will raise a KeyError for a nonexistent option. FWIW pandas raises a customized `OptionError` here. We don't have to mimic that, but we might want to provide a slightly more informative error using a `try-except` to indicate that the option doesn't exist rather than printing out that a key doesn't exist in the `_OPTIONS` dict (which is what will show up now).

Reply:
Done